### PR TITLE
Ensure ISO's defaults are applied

### PR DESCRIPTION
**Pull request by davidcassany**
Signed-off-by: David Cassany <dcassany@suse.com>

### CODE REVIEW

File: cmd/config/config.go [hunk 1]
```diff
@@ -54,7 +55,12 @@ func ReadConfigBuild(configDir string, mounter mount.Interface) (*v1.BuildConfig
 	viperReadEnv()
 
 	// unmarshal all the vars into the config object
-	err := viper.Unmarshal(cfg)
+	err := viper.Unmarshal(cfg, func(config *mapstructure.DecoderConfig) {
+		// Make sure we zero fields before applying them, this is relevant for slices
+		// so we do not merge with any already present value and directly apply whater
+		// we got form configs.
+		config.ZeroFields = true
```
Comment (davidcassany) [inline cmd/config/config.go:62]: This was driving me mad. It took me a while to figure out how to do this.

My default ISOs config (from `config.NewISO()`) included `[live/grub2 live/grub2-efi-iomage]` in `ISO.Image`, and my manifest was having

```
iso:
  image:
  - custom/grub2
```
Without the `ZeroField` flag the merge process was resulting to `[custom/grub2 live/grub2-efi-iomage]` 😞

File: cmd/config/config_test.go [hunk 2]
```diff
@@ -56,6 +56,8 @@ var _ = Describe("Config", func() {
 			Expect(cfg.Name).To(Equal("cOS-0"))
 			hasSuffix := strings.HasSuffix(viper.ConfigFileUsed(), "config/manifest.yaml")
 			Expect(hasSuffix).To(BeTrue())
+			Expect(len(cfg.ISO.Image)).To(Equal(1))
+			Expect(cfg.ISO.Image[0]).To(Equal("recovery/cos-img"))
```
Comment (davidcassany) [inline cmd/config/config_test.go:60]: This checks the `ZeroField` flag is applied and that the list form the manifest.yaml is not merged with the previous default one.

**Review by mudler — APPROVED** [review]
Looks good!

File: cmd/config/config.go (hunk 1 above)
Comment (mudler) [reply cmd/config/config.go:62]: I see, this indeed is far from obvious from viper usage...

Comment (Itxaka) [reply cmd/config/config.go:62]: wow, great catch

**Review by Itxaka — APPROVED** [review]

**Pull request merged**